### ISSUE TITLE
Implement comm diagnostics for network atomics

### ISSUE THREAD

**Issue by ronawho** [chapel-lang/chapel#10124]
We have communication counters/diagnostics for things like gets, puts and on-stmts. These are really valuable for tracking the amount of communication that occurs in specific parts of code since that's not always obvious from the source code.

For similar reasons we should add diagnostics for network atomics, which is something we don't currently track.

**Comment by ronawho** [comment]
From https://github.com/chapel-lang/chapel/issues/11811#issuecomment-445082458:

>AMOs might be useful to count, too. And we're starting to get a hierarchy: regular and fast executeOns can both be either small or large, and it might be nice to know how many AMOs were done natively vs. by AM-mediated processor atomics, at least on a per-op basis and maybe for convenience across all ops as well.

**Comment by ronawho** [comment]
> AMOs might be useful to count, too. And we're starting to get a hierarchy: regular and fast executeOns can both be either small or large, and it might be nice to know how many AMOs were done natively vs. by AM-mediated processor atomics, at least on a per-op basis and maybe for convenience across all ops as well.

Yeah, agreed. I don't think we'll want to have comm diags for every single OP, but maybe each class of OP (cas, read, write, fetching, non-fetching)

**Comment by gbtitus** [comment]
I'd be inclined to count and report them individually, based on not throwing away information prematurely. 
They can always be reduced into classes by postprocessing, after all.  Or there's probably a way to design the interface so you could get either one.
